optuna/importance/_mean_decrease_impurity.py

```diff
@@ -52,6 +52,9 @@ def __init__(
             min_samples_leaf=1,
             random_state=seed,
         )
+        self._trans_params = numpy.empty(0)
+        self._trans_values = numpy.empty(0)
+        self._param_names: List[str] = list()
```

Comment:
My minor concern comes from this change. In my understanding, `self. _trans_params` and `self._param_names` are stored to be used in the `evaluate` method in the shap class after computing the parent's `evaluate` method. Similar to the comment on the decomposition of random forest fitting, this part also might be split as a method or static method rather than storing them as instance attributes.